lte/gateway/c/oai/tasks/sctp/sctp_itti_messaging.c

```diff
@@ -28,39 +28,70 @@
 //------------------------------------------------------------------------------
 int sctp_itti_send_lower_layer_conf(
-    task_id_t origin_task_id, sctp_assoc_id_t assoc_id, sctp_stream_id_t stream,
-    uint32_t mme_ue_s1ap_id, bool is_success) {
+    task_id_t origin_task_id, sctp_ppid_t ppid, sctp_assoc_id_t assoc_id,
+    sctp_stream_id_t stream, uint32_t ap_id, bool is_success) {
   MessageDef* msg = itti_alloc_new_message(TASK_SCTP, SCTP_DATA_CNF);
 
-  SCTP_DATA_CNF(msg).assoc_id       = assoc_id;
-  SCTP_DATA_CNF(msg).stream         = stream;
-  SCTP_DATA_CNF(msg).mme_ue_s1ap_id = mme_ue_s1ap_id;
-  SCTP_DATA_CNF(msg).is_success     = is_success;
+  SCTP_DATA_CNF(msg).ppid     = ppid;
+  SCTP_DATA_CNF(msg).assoc_id = assoc_id;
+  SCTP_DATA_CNF(msg).stream   = stream;
+
+  if (ppid == S1AP) {
+    OAILOG_DEBUG(
+        LOG_SCTP, "ppid match S1AP in sctp_itti_send_new_message_ind ");
+    SCTP_DATA_CNF(msg).mme_ue_s1ap_id = ap_id;
+  } else if (ppid == NGAP) {
+    OAILOG_DEBUG(
+        LOG_SCTP, " ppid match NGAP in sctp_itti_send_lower_layer_conf ");
+    SCTP_DATA_CNF(msg).amf_ue_ngap_id = ap_id;
+  } else {
+    OAILOG_ERROR(
+        LOG_SCTP, "ppid: %d not matching in sctp_itti_send_lower_layer_conf ",
+        ppid);
+  }
+
+  SCTP_DATA_CNF(msg).is_success = is_success;
 
   return send_msg_to_task(&sctp_task_zmq_ctx, origin_task_id, msg);
 }
 
 //------------------------------------------------------------------------------
 int sctp_itti_send_new_association(
-    sctp_assoc_id_t assoc_id, sctp_stream_id_t instreams,
+    sctp_ppid_t ppid, sctp_assoc_id_t assoc_id, sctp_stream_id_t instreams,
     sctp_stream_id_t outstreams, STOLEN_REF bstring* ran_cp_ipaddr) {
   MessageDef* msg = itti_alloc_new_message(TASK_SCTP, SCTP_NEW_ASSOCIATION);
 
-  SCTP_NEW_ASSOCIATION(msg).assoc_id      = assoc_id;
-  SCTP_NEW_ASSOCIATION(msg).instreams     = instreams;
-  SCTP_NEW_ASSOCIATION(msg).outstreams    = outstreams;
-  SCTP_NEW_ASSOCIATION(msg).ran_cp_ipaddr = *ran_cp_ipaddr;
-
-  return send_msg_to_task(&sctp_task_zmq_ctx, TASK_S1AP, msg);
+  if (ppid == S1AP) {
+    SCTP_NEW_ASSOCIATION(msg).assoc_id      = assoc_id;
+    SCTP_NEW_ASSOCIATION(msg).instreams     = instreams;
+    SCTP_NEW_ASSOCIATION(msg).outstreams    = outstreams;
+    SCTP_NEW_ASSOCIATION(msg).ran_cp_ipaddr = *ran_cp_ipaddr;
+    OAILOG_DEBUG(
+        LOG_SCTP, "ppid match S1AP in sctp_itti_send_new_message_ind ");
+    return send_msg_to_task(&sctp_task_zmq_ctx, TASK_S1AP, msg);
+  } else if (ppid == NGAP) {
+    SCTP_NEW_ASSOCIATION(msg).assoc_id   = assoc_id;
+    SCTP_NEW_ASSOCIATION(msg).instreams  = instreams;
+    SCTP_NEW_ASSOCIATION(msg).outstreams = outstreams;
+    OAILOG_DEBUG(
+        LOG_SCTP, " ppid match NGAP in sctp_itti_send_new_association ");
+    return send_msg_to_task(&sctp_task_zmq_ctx, TASK_NGAP, msg);
+  } else {
+    OAILOG_ERROR(
+        LOG_SCTP, "ppid: %d not matching in sctp_itti_send_new_association ",
+        ppid);
+    return 0;
```

Comment:
Use the macros `RETURNok` or `RETURNerror` instead of numerals in return code

Reply:
okay